src/App.js

```diff
@@ -4,26 +4,41 @@ import './App.css';
 import chatMessages from './data/messages.json';
 import ChatLog from './components/ChatLog';
 
-const App = () => {
-  const [likes, setLikes] = useState(0);
-  const incrementLikes = () => {
-    setLikes(likes + 1);
+const countLikes = (messages) => {
+  let count = 0;
+  for (const message of messages) {
+    if (message.liked) {
+      count += 1;
+    }
   }
-  const decrementLikes = () => {
-    setLikes(likes - 1);
+  return count;
+};
+
+const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const onLikeClicked = (id) => {
+    setMessages(prev => {
+      return prev.map(currentMessage => {
+        if (id === currentMessage.id) {
+          return {...currentMessage, liked: !currentMessage.liked}        
+        } else {
+          return currentMessage;
+        }
+      })
+    });
```

Comment:
👍 This addresses the requirements of upating state by
1. getting the most up-to-date copy of `messages` (using the callback style of `setMessage`)
2. making a new list (by using `map`) so that React will see the change (it compares lists by location, not value)
3. find the message that was clicked by comparing the id of each to the id passed by ChatEntry
4. replace the clicked message with a new copy, having its liked status flipped
5. allow unchanged messages to pass unchanged into the new list

This is very dense code, and relies on a lot of JS fundamentals, but it does tend to be one of the cleaner ways of achieving these steps. The more we write code like this, the easier it becomes to recognize what similar code is doing in other situations.